### ISSUE TITLE
[kubeadam][structured authz] can't use structured authz due to default `--authorization-mode` flag

### ISSUE THREAD

**Issue by LiorLieberman** [kubernetes/kubernetes#123651]
### What happened?

**TL;DR**
https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/cmd/kubeadm/app/phases/controlplane/manifests.go#L232 prevents us from using structured authz (setting `--authorization-config` flag) as those flags are mutually exclusive

/cc @palnabarun (as the author of https://github.com/kubernetes/kubernetes/pull/120154)
/cc @liggitt 

FYI @stealthybox @robscott 

To use structured authz you need to define the authorization configuration using the `--authorization-config` flag.

However this flag is mutually exclusive with `--authorization-modes` and `--authorization-webhook-*` flags.  (refer to [1](https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/pkg/kubeapiserver/options/authorization.go#L113) & [2](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file))

Kubeadam adds `--authorization-mode` by default ([1](https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/cmd/kubeadm/app/phases/controlplane/manifests.go#L232) & [2](https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/cmd/kubeadm/app/phases/controlplane/manifests.go#L241)) which prevents us from using `--authorization-config`

### What did you expect to happen?

I should be able to define the `authorization-config`  flag and have structured authz working.

### How can we reproduce it (as minimally and precisely as possible)?

Spin up a kind cluster using this kind config (make sure the images are > 1.29 for structured authz to be avaliable)

```yaml
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
featureGates:
  StructuredAuthorizationConfiguration: true
kubeadmConfigPatches:
  - |
    kind: ClusterConfiguration
    metadata:
      name: config
    apiServer:
      extraArgs:
        authorization-config: "/files/authorization_config.yaml" 
      extraVolumes:
      - name: api-server-basic-auth-files
        hostPath: "/files"
        mountPath: "/files"
        readOnly: true
nodes:
- role: control-plane
  # add a mount from /path/to/my/files on the host to /files on the node
  extraMounts:
  - hostPath: <some-path-on-my-host>/refgrant-poc/files
    containerPath: /files
```

(For this bug, the content of `/files/authorization_config.yaml` doesn't matter.)

However, the cluster become unhealthy.
docker exec to the kind cluster container, where we can see the inner container of the apiserver has exited. 
Running `crictl logs <container>` we can see this error

<img width="1720" alt="image" src="https://github.com/kubernetes/kubernetes/assets/76940994/efeee626-7b6d-42de-a5e8-cee546b60194">

### Kubernetes version

<details>

```console
The kind cluster version image: `kindest/node:v1.29.2`
```

</details>

### Cloud provider

<details>
N/A -I am using kind
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by LiorLieberman** [comment]
@BenTheElder FYI, in case someone opens similar issue in Kind. I might be wrong but I think this effectively means that kind users cant use structured authz (unless it supports a mode to spin up a cluster without kubeadam).
